src/main/java/uk/gov/digital/ho/hocs/audit/core/RequestData.java

```diff
@@ -9,26 +9,59 @@
 import javax.servlet.http.HttpServletResponse;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
 import java.util.UUID;
 
 @Component
 public class RequestData implements HandlerInterceptor {
 
-    public static final String CORRELATION_ID_HEADER = "X-Correlation-Id";
-    public static final String USER_ID_HEADER = "X-Auth-UserId";
-    public static final String USERNAME_HEADER = "X-Auth-Username";
-    public static final String USER_ROLES_HEADER = "X-Auth-Roles";
+    static final String CORRELATION_ID_HEADER = "X-Correlation-Id";
+    static final String USER_ID_HEADER = "X-Auth-UserId";
+    static final String USERNAME_HEADER = "X-Auth-Username";
+    static final String GROUP_HEADER = "X-Auth-Groups";
+    static final String USER_ROLES_HEADER = "X-Auth-Roles";
 
     private static final String ANONYMOUS = "anonymous";
     private static final String BLANK = "";
 
+    private static boolean isNullOrEmpty(String value) {
+        return value == null || value.equals("");
+    }
+
+    public void parseMessageHeaders(Map<String, String> headers) {
+        headers.forEach(MDC::put);
```

Comment:
This needed if you do the below. Unsure you need both?

Reply:
I think were confusing ourselves, you have `headers.forEach(MDC::put);` but then do more puts if they exist - of which they would already be in the map? Unless your trying to keep the Queues named interpretation which I don't think we need.